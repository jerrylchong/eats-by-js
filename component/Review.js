import React, {useState, useEffect} from 'react';
import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
import {getUsername} from '../helpers/apiHelpers'

const Review = (props) => {
    const {user_id, date, title, rating, content} = props
    const [user, setUser] = useState("");
    const ratingContainerStyle = {
        backgroundColor: parseFloat(rating) < 0
            ? '#b3b3b3'
            : parseFloat(rating) < 2.5
                ? '#d68c6c'
                : parseFloat(rating) < 4
                    ? '#ffbf00'
                    : '#66b58c',
        flexDirection: 'row',
        height: Dimensions.get('window').width * 0.15,
        width: Dimensions.get('window').width * 0.15,
        borderRadius: 36,
        justifyContent: 'space-evenly',
        alignItems: 'center'
    }
    useEffect(() => {
        getUsername(user_id).then(data => setUser(data.username))
    },[]);
    return (
        <View style = {styles.container}>
            <View style={styles.avatar}>
                <View style = {ratingContainerStyle}>
                    {parseFloat(rating) < 0
                        ? <Text style = {styles.rating}>NA</Text>
                        : parseFloat(rating) < 2.5
                            ? <Image style = {styles.face} source={require('../assets/facebad.png')}/>
                            : parseFloat(rating) < 4
                                ? <Image style = {styles.face} source={require('../assets/facemeh.png')}/>
                                : <Image style = {styles.face} source={require('../assets/facegood.png')}/>}
                </View>
            </View>
            <View style={styles.rightColumn}>
                <View style = {styles.titleRating}>
                    <Text style = {styles.title}>{title}</Text>
                    <Text style = {styles.rating}>{rating}/5</Text>
                </View>
                <Text style = {styles.user}>{date} . {user}</Text>
                <Text style = {styles.content}>{content}</Text>
            </View>
        </View>
    )
}

export default Review

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: Dimensions.get('window').height * 0.1,
        justifyContent: 'flex-start',
        borderBottomWidth: 0.5,
        borderColor: '#b3b3b3',
        paddingHorizontal: '3%',
        flexDirection: 'row',

    },
    rightColumn: {
        width:"80%",
        marginLeft:"5%",
        flexDirection: 'column',
        justifyContent: 'center',
        height: "100%",
    },
    avatar: {
        height: "100%",
        justifyContent: 'center',
    },
    titleRating: {
        height: '30%',
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    title: {
        flex:10,
        color: '#404040',
        fontSize: 16,
        fontFamily: 'Ubuntu-Bold'
    },
    rating: {
        flex:1,
        color: '#404040',
        fontSize: 14,
        fontFamily: 'Ubuntu-Bold'
    },
    user: {
        color: '#b3b3b3',
        fontSize: 12,
        fontFamily: 'Ubuntu-Medium'
    },
    content: {
        color: '#b3b3b3',
        fontSize: 12,
        flexWrap: 'wrap',
        fontFamily: 'Ubuntu'
    },
    face: {
        height: Dimensions.get('window').width * 0.15 - 8,
        width: Dimensions.get('window').width * 0.15 - 8,
    },
})
