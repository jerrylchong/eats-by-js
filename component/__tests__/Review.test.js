import React from 'react';
import renderer from 'react-test-renderer';
import Review from "../Review";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Text, Image} from "react-native";

// snapshot test
test('renders correctly', () => {
    const tree = renderer.create(<Review/>).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });

test('test component rendering', () => {
    const wrapper = shallow(
        <Review
            title={'title'}
            rating={'5'}
            date={'date'}
            content={'content'}
        />);

    expect(wrapper.childAt(0).childAt(0).find(Image)).toBeTruthy(); // checks if title text is rendered
    expect(wrapper.childAt(1).childAt(0).find(Text).at(0).dive().text()).toEqual('content'); // checks if rating text is rendered
    expect(wrapper.childAt(1).childAt(0).find(Text).at(1).dive().text()).toEqual('5/5'); // checks if rating text is rendered
    expect(wrapper.childAt(1).find(Text).at(2)).toBeTruthy(); // checks if rating text is rendered
})