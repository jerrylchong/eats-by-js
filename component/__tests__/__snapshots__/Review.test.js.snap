// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0.5,
      "borderColor": "#b3b3b3",
      "flexDirection": "row",
      "height": 133.4,
      "justifyContent": "flex-start",
      "paddingHorizontal": "3%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#66b58c",
          "borderRadius": 36,
          "flexDirection": "row",
          "height": 112.5,
          "justifyContent": "space-evenly",
          "width": 112.5,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 104.5,
            "width": 104.5,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
        "marginLeft": "5%",
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": "70%",
          "justifyContent": "space-between",
          "paddingTop": "3%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#404040",
            "flexWrap": "wrap",
            "fontFamily": "Ubuntu-Bold",
            "fontSize": 16,
            "width": "90%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#404040",
            "fontFamily": "Ubuntu-Bold",
            "fontSize": 14,
          }
        }
      >
        /5
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#b3b3b3",
          "flexWrap": "wrap",
          "fontFamily": "Ubuntu",
          "fontSize": 12,
          "marginBottom": "2%",
        }
      }
    >
      
    </Text>
  </View>
</View>
`;
